packages/api/src/bsky-agent.ts

```diff
@@ -609,3 +645,103 @@ async function updateFeedPreferences(
   })
   return res
 }
+
+/**
+ * A helper specifically for updating muted words preferences
+ */
+async function updateMutedWords(
+  agent: BskyAgent,
+  mutedWords: AppBskyActorDefs.MutedWord[],
+  action: 'upsert' | 'update' | 'remove',
+) {
+  await updatePreferences(agent, (prefs: AppBskyActorDefs.Preferences) => {
+    let mutedWordsPref = prefs.findLast(
+      (pref) =>
+        AppBskyActorDefs.isMutedWordsPref(pref) &&
+        AppBskyActorDefs.validateMutedWordsPref(pref).success,
+    )
+
+    if (mutedWordsPref && AppBskyActorDefs.isMutedWordsPref(mutedWordsPref)) {
+      if (action === 'upsert' || action === 'update') {
+        outer: for (const newItem of mutedWords) {
+          for (const existingItem of mutedWordsPref.items) {
+            if (existingItem.value === newItem.value) {
+              existingItem.targets =
+                action === 'upsert'
+                  ? Array.from(
+                      new Set([...existingItem.targets, ...newItem.targets]),
+                    )
+                  : newItem.targets
+              continue outer
```

Comment:
i'll be honest, i never use this syntax. does this accomplish the same thing as a `break` in the inner loop? and if so could we use that?

Reply:
Oh sure sure